### PR TITLE
SLLS-76 Subscribe for server events

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java [hunk 1]
```diff
@@ -57,25 +74,30 @@ public void initialize(@Nullable List<WorkspaceFolder> workspaceFolders) {
 
   public void didChangeWorkspaceFolders(WorkspaceFoldersChangeEvent event) {
     LOG.debug("Processing didChangeWorkspaceFolders event");
+    var removedFolderWrappers = new ArrayList<WorkspaceFolderWrapper>();
+    var addedFolderWrappers = new ArrayList<WorkspaceFolderWrapper>();
     for (var removed : event.getRemoved()) {
       var uri = create(removed.getUri());
-      removeFolder(uri);
+      removedFolderWrappers.add(removeFolder(uri));
```
Comment (henryju) [inline src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java:81]: Possibly adding `null` to the list, not sure this is good

File: src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java [hunk 2]
```diff
@@ -57,25 +74,30 @@ public void initialize(@Nullable List<WorkspaceFolder> workspaceFolders) {
...
       var uri = create(added.getUri());
       var addedWrapper = addFolder(added, uri);
+      addedFolderWrappers.add(addedWrapper);
       listeners.forEach(l -> l.added(addedWrapper));
     }
+    executor.submit(() -> bindingManager.subscribeForServerEvents(addedFolderWrappers, removedFolderWrappers));
   }
 
-  private void removeFolder(URI uri) {
+  private WorkspaceFolderWrapper removeFolder(URI uri) {
```
Comment (henryju) [inline src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java:92]: `@CheckForNull`

**Review by henryju — APPROVED** [review]
Small glitch then LGTM

File: src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java (hunk 1 above)
Comment (damien-urruty-sonarsource) [reply src/main/java/org/sonarsource/sonarlint/ls/folders/WorkspaceFoldersManager.java]: :+1:

**Pull request merged**